numba_cuda/numba/cuda/compiler.py

```diff
@@ -971,6 +972,173 @@ def define_error_gv(postfix):
     return helper_func
 
 
+def compile_all(
+    pyfunc,
+    sig,
+    debug=None,
+    lineinfo=False,
+    device=True,
+    fastmath=False,
+    cc=None,
+    opt=None,
+    abi="c",
+    abi_info=None,
+    output="ltoir",
+    forceinline=False,
+    launch_bounds=None,
+):
+    """Return a list of PTX/LTO-IR for kernel as well as external functions depended by the kernel.
+    If external functions are cuda c/c++ source, they will be compiled with NVRTC. The output code
+    kind is the same as the `output` parameter.
+    Otherwise, they will be passed through to the return list.
+    """
+
+    if output not in ("ptx", "ltoir"):
+        raise NotImplementedError(f"Unsupported output type: {output}")
+
+    if forceinline and output != "ltoir":
+        raise ValueError("Can only designate forced inlining in LTO-IR")
```

Comment:
Does this need to be a hard error? Is `forceinline` a guarantee or can the compiler still ignore it?

Reply:
That's correct - you can't inline PTX. This looks like new code on the diff, but this PR is only really moving it from the original `compile()` implementation.